packages/react-instantsearch-dom-maps/src/GoogleMapsContext.ts

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+export type GoogleMapsContextState = {
+  google: typeof google | null;
```

Comment:
Can you elaborate?

Mounting the `Provider` only when the map is ready will change the component's lifecycle.

Reply:
Right now we mount the `Provider` even when the map is not ready. We don't have to because the children that consume the value are only mounted once the map is ready. We probably can do the same with the `Provider`.

https://github.com/algolia/react-instantsearch/blob/d4c71892efb9d1dbd86544d7bf5c22e6fe0548d8/packages/react-instantsearch-dom-maps/src/GoogleMaps.js#L138-L143

Reply:
Right now we rely on `componentDidMount` to set the Google Maps instance based on the map ref:

https://github.com/algolia/react-instantsearch/blob/d4c71892efb9d1dbd86544d7bf5c22e6fe0548d8/packages/react-instantsearch-dom-maps/src/GoogleMaps.js#L35

Mounting `Provider` only when the map is ready means that we don't have access to this ref anymore.

In order for this to work, we'd need to change the whole component's lifecycle. Is that something that we want to address in this change?

Reply:
I'm not sure to understand the argument of the `ref`. What prevent us to do something like this?

```jsx
<div className={cx("")}>
  <div ref={this.mapRef} className={cx("map")} />
  {isMapReady && (
    <GoogleMapsContext.Provider value={this.state.value}>
      {children}
    </GoogleMapsContext.Provider>
  )}
</div>
```

Reply:
Ah, now I understand.